test/cppsuite/test_harness/workload/database_model.h

```diff
@@ -56,18 +56,37 @@ struct collection_t {
 /* Representation of the collections in memory. */
 class database {
     public:
-    const std::vector<std::string>
-    get_collection_names() const
+    std::mutex &
+    get_mtx()
+    {
+        return (_mtx);
+    }
+
+    /*
+     * The database's methods take a `std::lock_guard`. We don't do anything with the guard itself,
+     * but its a way to make sure that callers don't forget to lock and accidentally shoot
+     * themselves in the foot.
+     */
+    std::vector<std::string>
+    get_collection_names(const std::lock_guard<std::mutex> &) const
     {
         std::vector<std::string> collection_names;
 
-        for (auto const &it : collections)
+        for (auto const &it : _collections)
             collection_names.push_back(it.first);
 
         return (collection_names);
     }
 
-    std::map<std::string, collection_t> collections;
+    std::map<std::string, collection_t> &
+    get_collections(const std::lock_guard<std::mutex> &)
```

Comment:
Same question here for the "const" keyword

Reply:
Don't think that would work. The function is returning a non-const ref to a member.